### PR TITLE
Map block: Allow block selection when Mapkit is being used

### LINKED ISSUE #31870: Map block: Selecting the Map block in the editor is very difficult
### Impacted plugin

Jetpack

### Quick summary

Clicking on a Map block causes you to interact with the map itself, rather than the block, which makes it nearly impossible to select the block without using the List View or the "select" tool.

### Steps to reproduce

1. Insert a Map block on a post or page.
2. Click on any other block (or the post/page title).
3. Click on the Map block again.

### A clear and concise description of what you expected to happen.

The Map block should be selected, with the block toolbar visible, so that it can be edited, moved, or deleted.

### What actually happened

Focus doesn't seem to move to the Map block and remains on the previously selected element. Clicking around the Map block interacts with the map itself.

https://github.com/Automattic/jetpack/assets/5630717/59332383-f955-4143-8ded-ab28de41b9ff

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Atomic

### Logs or notes

Only replicated on WoA so far, but I imagine it affects Simple too.

Workarounds:

* Use the List View to select the block.
* Use the "select" tool (instead of "edit") to click on the block.
* Depending on content layout, you can click on the block above/below the Map block, then tap down/up on your keyboard (respectively) and move your cursor over the map to cause the block toolbar to display.

### PR DESCRIPTION
**Pull request by coder-karen**
Fixes https://github.com/Automattic/jetpack/issues/31870

## Proposed changes:

* This PR adds an overlay div to the Mapkit map markup that is rendered, in order to allow the block to be selectable.
* This fixes an issue where it isn't possible to use a mouse to select the block just by clicking on the block, within the wp-admin post / page editor.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/31870

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

On self-hosted:

* Though this isn't an issue for self-hosted sites as they use Mapbox for the maps used in the Map block, you can still replicate the issue and test the fix there. 
* When adding a Map block to a post or page, add the following via the developer console: `document.cookie = "map_provider=mapkit";`. You should then be able to add a map using Mapkit. Make sure the map style is not 'Terrain' as that uses Mapbox.
* On trunk, notice that you cannot use your mouse to select the block (the map block settings are not visible in the sidebar, you can only move the map).
* With this PR applied, you will be able to see the map settings in the sidebar. With an extra mouse click, you can also move around the map as before.

On WoA / Simple:
* Add a Map block to a post or page in wp-admin, not WordPress.com (eg. `/wp-admin/post.php?post={postnumber}&action=edit`)., and without the PR applied you should be able to replicate the issue as above. Make sure the map style is not 'Terrain' as that uses Mapbox.
* On WoA apply the PR via the beta tester plugin, and on Simple you can use the command in the generated comment below to apply the changes to a sandboxed site.
* You should now be able to select the Map block itself and see the map settings, just by clicking on the map. An additional click should allow you to move around the map.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/map/component/mapkit.js [hunk 1]
```diff
@@ -80,11 +106,25 @@ const MapkitComponent = forwardRef(
 						onMapLoaded={ onMapLoaded }
 					/>
 				) : null }
+
+				{ /* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */ }
```
Comment (coder-karen) [inline projects/plugins/jetpack/extensions/blocks/map/component/mapkit.js:110]: Noting that I'm just focussed on mouse click events - this PR doesn't introduce other accessibility issues but I did notice existing ones that should get fixed later: https://github.com/Automattic/jetpack/issues/38986

**Review by darssen — COMMENTED** [review]
Although I am not able to replicate the issue, this doesn't seem to introduce any problems either. 
@jp-imagines can you test and check if this solves the issue for you?

**Comment by jp-imagines** [comment]
@darssen I was able to replicate the original issue on a test site again, and applying this PR did fix it – I can select the Map block again and interact with it on a second click.

**Review by darssen — APPROVED** [review]
Great! 
 :ship:

**Review by sergeymitr — APPROVED** [review]
I was only able to reproduce the issue on a self-hosted site using `document.cookie = "map_provider=mapkit";`, but can confirm that the PR solved the problem.
Great job 👍

**Pull request merged**